packages/core/http/core-http-server-internal/src/csp/config.ts

```diff
@@ -108,6 +108,21 @@ const configSchema = schema.object(
  */
 export type CspConfigType = TypeOf<typeof configSchema>;
 
+/**
+ * @internal
+ */
+export type CspAdditionalConfig = Pick<
+  Partial<CspConfigType>,
+  | 'connect_src'
+  | 'default_src'
+  | 'font_src'
+  | 'frame_src'
+  | 'img_src'
+  | 'script_src'
+  | 'style_src'
+  | 'worker_src'
+>;
```

Comment:
NIT: I would just have gone with `Partial<CspConfigType>`, but what you did is probably better 😄